src/simple_server/core.clj

```diff
@@ -2,43 +2,86 @@
   (:require [clojure.pprint]
             [ring.adapter.jetty :refer [run-jetty]]
 
-            [ring.util.response :refer [response created redirect not-found status]]
+            [ring.util.response :refer [response content-type created redirect not-found status]]
             [compojure.core :refer [GET POST ANY defroutes]]
             [compojure.coercions :refer [as-int]]
             [ring.middleware.defaults :as middleware]
 
             [ring.mock.request :as mock]
             [simple-server.simple-game :as game]))
 
-;;; Finally, let us truly separate concerns between our "application code"
-;;; and our "http code".  Our game now lives in its own namespace, and
-;;; is fully testable independent of our "presentation layer".
+(def active-logins (atom #{}))
 
-(defn new-game-handler []
-  (when (game/new-game!)
-    (response "OK- start guessing at /guess")))
+(defn web-content
+  "Returns a handler for web page display"
+  [text]
+  (-> text
+      (response)
+      (content-type "text/plain")))
 
-(defn guess-handler [guess]
-  (condp = (game/guess-answer guess)
-    nil       (-> (response  "You need to supply a guess with /guess?guess=N")
-                  (status 400))
-    :game-over (response  "Congratulations! You win!")
-    :too-low   (response "Too low.")
-    :too-high  (response  "Too high.")))
+(defn logged-in?
+  "Returns if ACCOUNT is logged in (exists)"
+  [account]
+  (some #{account} @active-logins))
+
+(defn new-game-handler 
+  "Starts a new game on ACCOUNT"
+  [account]
+  (if (logged-in? account)
```

Comment:
Nice attempt.  I have a few comments:

* Notice how logged-in? now has infiltrated both new-game-handler and guess-handler?  Can you think of a way to separate these concerns more nicely? Hint: think about middlewares.

* Do you think passing the account on the URL is the best way to do it?  Why or why not?  List some advantages/disadvantages to your approach.  How else might it have been done?